### ISSUE TITLE
Does the pcc module pass in parameter cls_i contain a private category?

### ISSUE THREAD

**Issue by yanguo-lucky** [zyfone/DPA#1]
Hello, I encountered some problems when reading the code. In the PCC module, the two parameters cls_i_s and cls_i_t were passed in. Later, these two parameters were used to extract the private category by the way that the shared class and the private class have no intersection. I saw that the category score was obtained by the fully connected layer. How can there be private categories in the two parameters cls_i_s and cls_i_t? How are they detected?

**Comment by zyfone** [comment]
Hello, thanks for your attention to our work.  

1. **How can there be private categories in the two parameters \( \text{cls\_i\_s} \) and \( \text{cls\_i\_t} \)?**  

   In the UniDAOD setting, the source domain contains private classes, but these are removed during testing. However, in the target domain, private classes are mistakenly classified as source domain classes (noise-common).  

2. **How are they detected?**  
   - \( \text{cls\_i\_s} \) consists of both **common** and **private classes**.  
   - \( \text{cls\_i\_t} \) consists of both **common** and **noise-common classes**.  

   **Note**: _The **noise-common label** corresponds to a common class, but the actual **instances** in image belongs to a private class in the target domain._  
By computing the intersection, we can derive pseudo labels for private classes. Our goal is to ensure \( s\_common = t\_common \), but this intersection should be smaller than the actual common class set to effectively reduce noise interference. If not, the PCC module will fail to function properly.   
`if len(index_t_p)>1 and len(index_s_p)>1:`

Although **private classes** may contain common classes, and **noise-common classes** may also contain common classes, this means that the class sets of source and target images are not entirely the same within the common class set in this batch.

**Comment by yanguo-lucky** [comment]
Thanks for your reply. I have a new understanding, but I have a new question. If this is the case, should common and noise-common classes be found in \text{cls_i_s}? Or in terms of a batch, even a public class may be judged as a private class in a batch.

**Comment by yanguo-lucky** [comment]
> Hello, thanks for your attention to our work.
> 
> 1. **How can there be private categories in the two parameters ( \text{cls_i_s} ) and ( \text{cls_i_t} )?**
>    In the UniDAOD setting, the source domain contains private classes, but these are removed during testing. However, in the target domain, private classes are mistakenly classified as source domain classes (noise-common).
> 2. **How are they detected?**
>    
>    * ( \text{cls_i_s} ) consists of both **common** and **private classes**.
>    * ( \text{cls_i_t} ) consists of both **common** and **noise-common classes**.
>    
>    **Note**: _The **noise-common label** corresponds to a common class, but the actual **instances** in image belongs to a private class in the target domain._
>    By computing the intersection, we can derive pseudo labels for private classes. Our goal is to ensure ( s_common = t_common ), but this intersection should be smaller than the actual common class set to effectively reduce noise interference. If not, the PCC module will fail to function properly.
>    `if len(index_t_p)>1 and len(index_s_p)>1:`
> 
> Although **private classes** may contain common classes, and **noise-common classes** may also contain common classes, this means that the class sets of source and target images are not entirely the same within the common class set in this batch.

Thanks for your reply. I have a new understanding, but I have a new question. If this is the case, should common and noise-common classes be found in \text{cls_i_s}? Or in terms of a batch, even a public class may be judged as a private class in a batch.There is also a pre-trained model that can provide resnet101, thank you very much

**Comment by zyfone** [comment]
1. In batch data (a source image and target images), even a public class could be considered private based on the current batch’s common class set. Due to gradient detachment in the source domain, treating a public class as private has no impact on the source domain. For the target domain, treating a public class as private depends on whether the source domain contains it. Therefore, PCC helps mitigate negative transfer. This is demonstrated in the following code:  

   ```python
   # gradient detachment
   F.mse_loss(graph_t, graph_s.detach())
   ```

2. A pre-trained model, such as ResNet101, is available at: [[DAOD_CODE v1.0](https://github.com/zyfone/DAOD_CODE/releases/tag/v1.0)](https://github.com/zyfone/DAOD_CODE/releases/tag/v1.0).

**Comment by yanguo-lucky** [comment]
> 1. 在批量数据（源图像和目标图像）中，即使是公共类也可以根据当前批次的公共类集被视为私有类。由于源域中的梯度分离，将公共类视为私有类不会对源域产生影响。对于目标域，将公共类视为私有类取决于源域是否包含它。因此，PCC 有助于减轻负迁移。以下代码演示了这一点：
>    # gradient detachment
>    F.mse_loss(graph_t, graph_s.detach())
> 2. 预先训练的模型（例如 ResNet101）可从以下位置获取：[ [DAOD_CODE v1.0](https://github.com/zyfone/DAOD_CODE/releases/tag/v1.0) ]（https://github.com/zyfone/DAOD_CODE/releases/tag/v1.0）。

Thank you for your reply, I understand, thank you very much
